### PR TITLE
[e2e] Install CNI using ClusterResourceSet and bump up CAPI to v0.3.9

### PR DESCRIPTION
**Pull request by sedefsavas**
What this PR does / why we need it:
This PR installs e2e CNIs using ClusterResourceSet.

This PR should hold until CAPI v0.3.9 release or whenever these changes are merged: kubernetes-sigs/cluster-api#3332

/hold

### CODE REVIEW

File: test/e2e/data/infrastructure-vsphere/cluster-template.yaml [hunk 1]
```diff
@@ -207,3 +209,22 @@ spec:
         kind: VSphereMachineTemplate
         name: '${ CLUSTER_NAME }'
       version: '${ KUBERNETES_VERSION }'
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "cni-${ CLUSTER_NAME }-crs-0"
+data: ${CNI_RESOURCES}
```
Comment (yastij) [inline test/e2e/data/infrastructure-vsphere/cluster-template.yaml:217]: I guess we should also export `CNI_RESOURCES` in `./hack/e2e.sh` ?

File: test/e2e/data/infrastructure-vsphere/cluster-template.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ kind: Cluster
 metadata:
   name: '${ CLUSTER_NAME }'
   namespace: '${ NAMESPACE }'
+  labels:
+    cni: "${ CLUSTER_NAME }-crs-0"
```
Comment (yastij) [inline test/e2e/data/infrastructure-vsphere/cluster-template.yaml:8]: If we start installing CNIs,  the `PodCIDR: 192.168.0.0/16` will conflict with `192.168.8.0/24` which is what we're using for machines

**Review by yastij — APPROVED** [review]
/approve

mostly lgtm, pending the comments

File: test/e2e/data/infrastructure-vsphere/cluster-template.yaml (hunk 2 above)
Comment (sedefsavas) [reply test/e2e/data/infrastructure-vsphere/cluster-template.yaml:8]: E2E is using `192.168.0.0/16` now, is this a recent change?

Comment (yastij) [reply test/e2e/data/infrastructure-vsphere/cluster-template.yaml:8]: @sedefsavas - sorry I missed this, it's not a recent all of our e2e VM subnets are located within that range.

**Review by yastij — APPROVED** [review]
/lgtm
/approve

File: test/e2e/data/infrastructure-vsphere/kustomization/cluster-template.yaml [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: '${ CLUSTER_NAME }'
+  name: '${ CLUSTER_NAME }'
+  namespace: '${ NAMESPACE }'
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/16
```
Comment (yastij) [inline test/e2e/data/infrastructure-vsphere/kustomization/cluster-template.yaml:13]: @sedefsavas - this should be `192.168.30.0/24` ?

Comment (sedefsavas) [reply test/e2e/data/infrastructure-vsphere/kustomization/cluster-template.yaml:13]: Opened a PR for this: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1029

File: test/e2e/data/infrastructure-vsphere/cluster-template.yaml (hunk 1 above)
Comment (sedefsavas) [reply test/e2e/data/infrastructure-vsphere/cluster-template.yaml]: Export happens in clusterctl framework `clusterctl.SetCNIEnvVar()`.

**Pull request merged**